src/types.jl

```diff
@@ -1,143 +1,219 @@
+
+            message = clean!(model)
+            isempty(message) || @warn message
+
+            return model
+        end
+
+    end |> eval
 
 end
 
 const Regressor =
-  Union{NeuralNetworkRegressor,MultitargetNeuralNetworkRegressor}
+    Union{NeuralNetworkRegressor, MultitargetNeuralNetworkRegressor}
+
+
+# Separately define the ImageClassifier
+mutable struct ImageClassifier{B, F, O, L} <: MLJFluxProbabilistic
+    builder::B
+    finaliser::F
+    optimiser::O
+    loss::L
+    epochs::Int
+    batch_size::Int
+    lambda::Float64
+    alpha::Float64
+    rng::Union{AbstractRNG, Int64}
+    optimiser_changes_trigger_retraining::Bool
+    acceleration::AbstractResource
+end
+
+function ImageClassifier(
+    ; builder::B = image_builder(VGGHack),
+    finaliser::F = Flux.softmax,
+    optimiser::O = Optimisers.Adam(),
+    loss::L = Flux.crossentropy,
+    epochs = 10,
+    batch_size = 1,
+    lambda = 0,
+    alpha = 0,
+    rng = Random.default_rng(),
+    optimiser_changes_trigger_retraining = false,
+    acceleration = CPU1(),
+) where {B, F, O, L}
+    model = ImageClassifier{B, F, O, L}(
+        builder,
+        finaliser,
+        optimiser,
+        loss,
+        epochs,
+        batch_size,
+        lambda,
+        alpha,
+        rng,
+        optimiser_changes_trigger_retraining,
+        acceleration,
+    )
+
+    message = clean!(model)
+    isempty(message) || @warn message
+
+    return model
+end
+
 
 MMI.metadata_pkg.(
-  (
-      NeuralNetworkRegressor,
-      MultitargetNeuralNetworkRegressor,
-      NeuralNetworkClassifier,
-      ImageClassifier,
-      NeuralNetworkBinaryClassifier,
-  ),
-    name="MLJFlux",
-    uuid="094fc8d1-fd35-5302-93ea-dabda2abf845",
-    url="https://github.com/alan-turing-institute/MLJFlux.jl",
-    julia=true,
-    license="MIT",
+    (
+        NeuralNetworkRegressor,
+        MultitargetNeuralNetworkRegressor,
+        NeuralNetworkClassifier,
+        ImageClassifier,
+        NeuralNetworkBinaryClassifier,
+    ),
+    name = "MLJFlux",
+    uuid = "094fc8d1-fd35-5302-93ea-dabda2abf845",
+    url = "https://github.com/alan-turing-institute/MLJFlux.jl",
+    julia = true,
+    license = "MIT",
 )
 
+const MODELSUPPORTDOC = """
+This model supports categorical columns in the input table. If present, such columns are embedded into
+dense vectors by the use of an additional `EntityEmbedder` layer after the input 
+as described in Entity Embeddings of Categorical Variables by Cheng Guo, Felix Berkhahn.
+"""
+
+const XDOC = """
+- `X` is either a `Matrix` or any table of input features (eg, a `DataFrame`) whose columns are of scitype
+  `Finite`; check column scitypes with `schema(X)`. If `X` is a `Matrix`, it is assumed to have columns corresponding to features and rows corresponding to observations. 
+  In case any `Multiclass` or `OrderedFactor` columns appear in `X`, the constructed network will use an `EntityEmbedder` layer to transform them into dense vectors.
+"""
```

Comment:
I'm not sure, but it looks like matrices are only supported if all features are continuous. If so, make this clear in the docs. If not, be sure you have a test.

Reply:
Yeah it's true. If it's matrix then must be continuous.

Reply:
Now I have
```
- `X` is either a `Matrix` or any table of input features (eg, a `DataFrame`) whose columns are of scitype
  `Continuous` or `Multiclass` or `OrderedFactor`; check column scitypes with `schema(X)`. 
  If `X` is a `Matrix`, it is assumed to be purely continuous and to have columns corresponding to features and rows 
  corresponding to observations. 
  In case any `Multiclass` or `OrderedFactor` columns appear in `X`, the constructed network will use an `EntityEmbedder` layer to transform them into dense vectors. 
```

Reply:
Done.